src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+        struct stat fileStat;
+        if (stat(centralStoreFile.c_str(), &fileStat) == 0) {
+            if (fileStat.st_mode & S_IFREG) {
+                string host = workerSocketPair[0];
+                int port = stoi(workerSocketPair[1]);
+                int workerGraphID = stoi(workerSocketPair[2]);
+                int dataPort = stoi(workerSocketPair[3]);
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+                if (port == thisWorkerPort) {
+                    continue;
+                }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionID = (data);
-            partitionID = Utils::trim_copy(partitionID, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionID, "info");
+                bool result = true;
+                std::cout << pthread_self() << " host : " << host << " port : " << port << " DPort : " << dataPort
+                          << std::endl;
+                int sockfd;
+                char data[INSTANCE_DATA_LENGTH + 1];
+                bool loop = false;
+                socklen_t len;
+                struct sockaddr_in serv_addr;
+                struct hostent *server;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+                sockfd = socket(AF_INET, SOCK_STREAM, 0);
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string workerList = (data);
-            workerList = Utils::trim_copy(workerList, " \f\n\r\t\v");
-            instance_logger.log("Received Worker List " + workerList, "info");
-
-            std::vector<string> workerSockets;
-            stringstream wl(workerList);
-            string intermediate;
-            while (getline(wl, intermediate, ',')) {
-                workerSockets.push_back(intermediate);
-            }
+                if (sockfd < 0) {
+                    instance_logger.log("Cannot create socket", "error");
+                    return 0;
+                }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+                if (host.find('@') != std::string::npos) {
+                    host = Utils::split(host, '@')[1];
+                }
 
-            JasmineGraphHashMapLocalStore graphDB;
-            JasmineGraphHashMapCentralStore centralDB;
+                server = gethostbyname(host.c_str());
+                if (server == NULL) {
+                    instance_logger.log("ERROR, no host named ", "error");
```

Comment:
Need to improve this error message. Need to give the host name which the error message is referring to.

Reply:
Replaced all similar occurrences with `instance_logger.error("ERROR, no host named " + host);`

Reply:
Ok